t.go

```diff
@@ -21,3 +23,87 @@ type T interface {
 	Skipf(format string, args ...interface{})
 	Skipped() bool
 }
+
+// RunT brings in more of the *testing.T API including Run and Parallel which are needed
+// for matrix tests.
+// *testing.T satisfies the RunT interface.
```

Comment:
Why not add `Run` and `Parallel` to `T`?  Why a new interface?

I get why you might name an interface `T`, because you're thinking of `testing.T`, even though it doesn't follow conventions for naming interfaces.  But I was confused by "runT" until I saw this.  If it had been named say "testRunner" then I'd assume it was an interface with "Run" method right away.

Reply:
We sometimes create T from ginkgo.
Hmm, that's richer than I expected: https://pkg.go.dev/github.com/onsi/ginkgo/v2#GinkgoTInterface

Based on that, I think we can add Parallel and Fail to ntest.T

That still leaves Run as unsupported and that's why this PR is complicated.